projects/packages/my-jetpack/_inc/components/welcome-flow/index.tsx

```diff
@@ -0,0 +1,137 @@
+import { Container, Col, Button } from '@automattic/jetpack-components';
+import { __ } from '@wordpress/i18n';
+import { close } from '@wordpress/icons';
+import clsx from 'clsx';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import useEvaluationRecommendations from '../../data/evaluation-recommendations/use-evaluation-recommendations';
+import isJetpackUserNew from '../../data/utils/is-jetpack-user-new';
+import useWelcomeBanner from '../../data/welcome-banner/use-welcome-banner';
+import useAnalytics from '../../hooks/use-analytics';
+import useMyJetpackConnection from '../../hooks/use-my-jetpack-connection';
+import { CardWrapper } from '../card';
+import ConnectionStep from './ConnectionStep';
+import EvaluationProcessingStep from './EvaluationProcessingStep';
+import EvaluationStep, { EvaluationAreas } from './EvaluationStep';
+import styles from './style.module.scss';
+import type { FC, PropsWithChildren } from 'react';
+
+const WelcomeFlow: FC< PropsWithChildren > = ( { children } ) => {
+	const { recordEvent } = useAnalytics();
+	const { dismissWelcomeBanner } = useWelcomeBanner();
+	const { submitEvaluation, saveEvaluationResult } = useEvaluationRecommendations();
+	const {
+		siteIsRegistered,
+		siteIsRegistering,
+		isUserConnected,
+		isSiteConnected,
+		handleRegisterSite,
+	} = useMyJetpackConnection( {
+		skipUserConnection: true,
+	} );
+	const [ isProcessingEvaluation, setIsProcessingEvaluation ] = useState( false );
+	const [ prevStep, setPrevStep ] = useState( '' );
+
+	const currentStep = useMemo( () => {
+		if ( ! siteIsRegistered ) {
+			return 'connection';
+		} else if ( ! isProcessingEvaluation ) {
+			if ( ! isJetpackUserNew() ) {
+				// If the user is not new, we don't show the evaluation step
+				return null;
+			}
+			return 'evaluation';
+		}
+
+		return 'evaluation-processing';
+	}, [ isProcessingEvaluation, siteIsRegistered ] );
+
+	useEffect( () => {
+		if ( prevStep !== currentStep ) {
+			recordEvent( 'jetpack_myjetpack_welcome_banner_step_view', { currentStep } );
```

Comment:
We should avoid uppercase letters in Tracks events since they don't match the pattern allowed for property names/values

```suggestion
			recordEvent( 'jetpack_myjetpack_welcome_banner_step_view', { current_step: currentStep } );
```

Reply:
+1 the upper case letters will break the tracks event